### PR TITLE
chore: unwrap single-element array in extractPresentationsFromVpToken

### PR DESCRIPTION
**Pull request by sanderPostma**
To satisfy PEX rules

### CODE REVIEW

File: packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts [hunk 1]
```diff
@@ -279,7 +281,7 @@ export const assertValidVerifiablePresentations = async (args: {
     hasher?: Hasher
   }
 }) => {
-  const presentationsArray = Array.isArray(args.presentations) ? args.presentations : [args.presentations]
+  const presentationsArray = args.presentations ? (Array.isArray(args.presentations) ? args.presentations : [args.presentations]) : []
```
Comment (nklomp) [inline packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts:284]: Presentations should always be present. See remark above. If you want to make a change then it should be an assertion we actually have at least one Presentation and fix the calling code. No Presentation doesn't exist with oid4vp. There is always at least one

**Review by nklomp — CHANGES_REQUESTED** [review]

File: packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts [hunk 2]
```diff
@@ -71,8 +71,7 @@ export const verifyPresentations = async (
   verifyOpts: VerifyAuthorizationResponseOpts,
 ): Promise<VerifiedOpenID4VPSubmission | null> => {
   const presentations = authorizationResponse.payload.vp_token
-    ? await extractPresentationsFromVpToken(authorizationResponse.payload.vp_token, { hasher: verifyOpts.hasher })
-    : []
+    ? await extractPresentationsFromVpToken(authorizationResponse.payload.vp_token, { hasher: verifyOpts.hasher }) : undefined
```
[the remark this answers is not in the document]
Comment (sanderPostma) [reply packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts:74]: fixed

File: packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts (hunk 1 above)
Comment (sanderPostma) [reply packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts:284]: fixed

File: packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts [hunk 3]
```diff
@@ -70,8 +70,11 @@ export const verifyPresentations = async (
   authorizationResponse: AuthorizationResponse,
   verifyOpts: VerifyAuthorizationResponseOpts,
 ): Promise<VerifiedOpenID4VPSubmission | null> => {
-  const presentations = authorizationResponse.payload.vp_token
-    ? await extractPresentationsFromVpToken(authorizationResponse.payload.vp_token, { hasher: verifyOpts.hasher }) : undefined
+  if (!authorizationResponse.payload.vp_token || Array.isArray(authorizationResponse.payload.vp_token) && authorizationResponse.payload.vp_token.length === 0) {
+    return Promise.reject('the payload is missing a vp_token')
```
Comment (nklomp) [inline packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts:74]: Promise.reject(Error('

File: packages/siop-oid4vp/lib/authorization-response/AuthorizationResponse.ts [hunk 4]
```diff
@@ -214,7 +214,11 @@ export class AuthorizationResponse {
     let nonce: string | undefined = this._payload.nonce
     if (this._payload?.vp_token) {
       const presentations = this.payload.vp_token ? await extractPresentationsFromVpToken(this.payload.vp_token, opts) : []
-      const presentationsArray = presentations ? (Array.isArray(presentations) ? presentations : [presentations]) : []
+      if (!presentations || (Array.isArray(presentations) && presentations.length === 0)) {
+        return Promise.reject('missing presentation(s)')
```
Comment (nklomp) [inline packages/siop-oid4vp/lib/authorization-response/AuthorizationResponse.ts:218]: Promise.reject(Error('

File: packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts [hunk 5]
```diff
@@ -105,8 +105,10 @@ export const verifyPresentations = async (
     return null
   }
 
-  const presentationsArray = presentations ? (Array.isArray(presentations) ? presentations : [presentations]) : []
-
+  if (!presentations || (Array.isArray(presentations) && presentations.length === 0)) {
+    return Promise.reject('missing presentation(s)')
```
Comment (nklomp) [inline packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts:109]: Promise.reject(Error('

File: packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts [hunk 6]
```diff
@@ -284,7 +286,11 @@ export const assertValidVerifiablePresentations = async (args: {
     hasher?: Hasher
   }
 }) => {
-  const presentationsArray = args.presentations ? (Array.isArray(args.presentations) ? args.presentations : [args.presentations]) : []
+  const {presentations} = args
+  if (!presentations || (Array.isArray(presentations) && presentations.length === 0)) {
+    throw Error('missing presentation(s)')
```
Comment (nklomp) [inline packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts:291]: Promise.reject(Error('

Comment (sanderPostma) [reply packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts]: Also fixed, I had to add a return type : Promise<void> to allow return Promise.reject()

File: packages/siop-oid4vp/lib/__tests__/IT.spec.ts [hunk 7]
```diff
@@ -141,7 +141,7 @@ function getVCs(): IVerifiableCredential[] {
   return vcs
 }
 
-describe('RP and OP interaction should', () => {
+describe.skip('RP and OP interaction should', () => {
```
Comment (nklomp) [inline packages/siop-oid4vp/lib/__tests__/IT.spec.ts:144]: Please do not just skip tests without making a reference to a ticket

File: packages/siop-oid4vp/lib/__tests__/SdJwt.spec.ts [hunk 8]
```diff
@@ -84,7 +84,7 @@ function getVCs(): OriginalVerifiableCredential[] {
   return [SD_JWT_VC]
 }
 
-describe('RP and OP interaction should', () => {
+describe.skip('RP and OP interaction should', () => {
```
Comment (nklomp) [inline packages/siop-oid4vp/lib/__tests__/SdJwt.spec.ts:87]: Please do not just skip tests without making a reference to a ticket

File: packages/siop-oid4vp/lib/__tests__/IT.spec.ts (hunk 7 above)
Comment (sanderPostma) [reply packages/siop-oid4vp/lib/__tests__/IT.spec.ts]: fixed

File: packages/siop-oid4vp/lib/__tests__/SdJwt.spec.ts (hunk 8 above)
Comment (sanderPostma) [reply packages/siop-oid4vp/lib/__tests__/SdJwt.spec.ts]: fixed

**Review by nklomp — APPROVED** [review]

**Pull request merged**